case-lib/pipeline.sh

```diff
@@ -1,7 +1,26 @@
 #!/bin/bash
 
+# This function will evaluate pipeline parameter related shell code generated by
+# - sof-tplgreader.py (for SOF, pipeline parameters dumped from topology)
+# - sof-dump-status.py (for legacy HDA, pipeline paramters dumped from proc)
+# Args: $1: SOF topology path
+#       $2: Pipeline filter in string form
+# Note: for legacy HDA, topology is not present, $1 will be empty.
 func_pipeline_export()
 {
+    # For legacy HDA platform, there is no topology, we have to export pipeline
+    # parameters from proc file system.
+    is_sof_used || {
+        filter_str="$2"
+        dlogi "No SOF sound card found, exporting pipeline parameters from proc file system"
+        tmp_pipeline_params=$(mktemp /tmp/pipeline-params.XXXXXXXX)
+        sof-dump-status.py -e "$filter_str" > "$tmp_pipeline_params" ||
+            die "Failed to export pipeline parameters from proc file system"
+        source "$tmp_pipeline_params" || return 1
+        rm "$tmp_pipeline_params"
+        return 0
+    }
+
```

Comment:
I think the `if [ $# -lt 1 ]; then...` check should be rephrased and kept at the start of the function.